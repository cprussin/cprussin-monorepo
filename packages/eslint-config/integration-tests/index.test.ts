import { exec } from "node:child_process";
import { rm } from "node:fs/promises";
import { join } from "node:path";
import { promisify } from "node:util";

const ONE_SECOND_IN_MS = 1000;

const TEST_DIR = join(__dirname, "__fixtures__", "test-package");
const NODE_MODULES = join(TEST_DIR, "node_modules");

const execAsync = promisify(exec);

describe("integration", () => {
  beforeEach(async () => {
    await rm(NODE_MODULES, { recursive: true, force: true });
  });

  afterEach(async () => {
    await rm(NODE_MODULES, { recursive: true });
  });

  it(
    "installs & works",
    async () => {
      await execAsync("pnpm --ignore-workspace i", { cwd: TEST_DIR });
      const { stdout } = await execAsync("pnpm exec eslint . || true", {
        cwd: TEST_DIR,
        env: { ...process.env, FORCE_COLOR: "0" },
      });
      expect(stdout.replace(TEST_DIR, "mocked-root-dir")).toMatchSnapshot();
    },
    120 * ONE_SECOND_IN_MS
  );
});
