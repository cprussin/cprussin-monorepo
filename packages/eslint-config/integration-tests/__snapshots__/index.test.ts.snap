// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`integration installs & works 1`] = `
"
mocked-root-dir/index.js
  1:1  error  Unexpected console statement  no-console

âœ– 1 problem (1 error, 0 warnings)

"
`;
